src/ape/managers/query.py

```diff
@@ -1,61 +1,19 @@
-from functools import partial
-from typing import Any, Dict, Optional
+from typing import Dict, Optional
 
 import pandas as pd
-from pydantic import BaseModel
 
 from ape.api import QueryAPI, QueryType
-from ape.api.query import BlockQuery, _BaseQuery
 from ape.exceptions import QueryEngineError
 from ape.plugins import clean_plugin_name
-from ape.utils import ManagerAccessMixin, cached_property, singledispatchmethod
-
-
-def get_columns_from_item(query: _BaseQuery, item: BaseModel) -> Dict[str, Any]:
-    return {k: v for k, v in item.dict().items() if k in query.columns}
-
-
-class DefaultQueryProvider(QueryAPI):
```

Comment:
Why is the `DefaultQueryProvider` being deleted?
I don't see it moved anywhere, or is it?

Are we no longer allowed to use the naive way?

Reply:
The idea is that `ape_cache.CacheQueryProvider`  is doing what `DefaultQueryProvider` did in that it provides a default implementation that brute force calls the data via `ProviderAPI` (hence why it is still a 1st class plugin) but also has the additional functionality of leveraging the cache database to speed up calls

Reply:
> If you do not want to cache, I will have to write in a check for update_cache() to prevent us from pushing new data to a caching db if that is not wanted

Got to thinking about this today, I think it should skip all the caching functionality if you purge the cache db and don't re-init it as a shortcircuit in case it's breaking behavior

Reply:
sounds good, I'll go back and work on that. Need to add quite a few tests to handle this

Reply:
Oh I now understand! It was to be the same as before + extra caching abilities, so not breaking in that sense.